server/admin_client.go

```diff
@@ -22,13 +22,18 @@ import (
 	"io/ioutil"
 	"net/http"
 
+	"github.com/cockroachdb/cockroach/client"
 	"github.com/cockroachdb/cockroach/util"
 )
 
 // sendAdminRequest send an HTTP request and processes the response for
 // its body or error message if a non-200 response code.
-func sendAdminRequest(req *http.Request) ([]byte, error) {
-	resp, err := http.DefaultClient.Do(req)
+func sendAdminRequest(ctx *Context, req *http.Request) ([]byte, error) {
+	client, err := client.NewHTTPClient(ctx.Certs)
```

Comment:
yeah. I want to rework the entire client setup, we should have only one initialized either at startup or lazily. TLS config is also wonky and needs to be reworked. I'll focus on getting cert generation which is needed for this PR, then do some refactoring.